internal/serialization/compact_field.go

```diff
@@ -67,37 +59,37 @@ var compactReaders = map[serialization.FieldKind]compactFieldReader{
 		return r.ReadArrayOfFloat64(field)
 	},
 	serialization.FieldKindString: func(r serialization.CompactReader, field string) any {
-		return readNullableCompactField(field, r.ReadString)
+		return r.ReadString(field)
 	},
 	serialization.FieldKindArrayOfString: func(r serialization.CompactReader, field string) any {
 		return r.ReadArrayOfString(field)
 	},
 	serialization.FieldKindDecimal: func(r serialization.CompactReader, field string) any {
-		return readNullableCompactField(field, r.ReadDecimal)
+		return r.ReadDecimal(field)
 	},
 	serialization.FieldKindArrayOfDecimal: func(r serialization.CompactReader, field string) any {
 		return r.ReadArrayOfDecimal(field)
 	},
 	serialization.FieldKindTime: func(r serialization.CompactReader, field string) any {
-		return readNullableCompactField(field, r.ReadTime)
+		return r.ReadTime(field)
 	},
 	serialization.FieldKindArrayOfTime: func(r serialization.CompactReader, field string) any {
 		return r.ReadArrayOfTime(field)
 	},
 	serialization.FieldKindDate: func(r serialization.CompactReader, field string) any {
-		return readNullableCompactField(field, r.ReadDate)
+		return r.ReadDate(field)
```

Comment:
AFAIS we removed nullable tag here but some of the other methods still contains it. Will we have different implementations for nullable?